data_structures/src/Stack/Stack.cs

```diff
@@ -26,36 +27,37 @@ public bool IsEmpty()
 
         public T Peek()
         {
-            if(!IsEmpty())            
-                return _items[_topOfStack -1];            
-            else            
-                throw new Exception("Stack is empty - Cannot peek.");            
+            if(IsEmpty())
```

Comment:
It's best to check your function's invariants first so that you can exit it as soon as possible.